constructors.go

```diff
@@ -58,7 +58,9 @@ func NewProviderGroup(name string, providers ...Provider) (Provider, error) {
 	opts := make([]YAMLOption, 0, len(providers)+2)
 	opts = append(opts, Name(name), Permissive())
 	for _, p := range providers {
-		opts = append(opts, Static(p.Get(Root).Value()))
+		if v := p.Get(Root); v.HasValue() {
+			opts = append(opts, Static(v.Value()))
+		}
```

Comment:
Since we don't have a reasonable way to pull it out of our internal code, we should un-deprecate this constructor. Let's remove the `Deprecated:` bit but keep a comment suggesting that people use `NewYAML` where possible.